datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -1297,6 +1298,32 @@ pub async fn from_substrait_rex(
                         outer_ref_columns,
                     })))
                 }
+                SubqueryType::SetPredicate(predicate) => {
+                    match predicate.predicate_op() {
+                        // exist
+                        PredicateOp::Exists => {
+                            let relations = &predicate.tuples;
+                            let plan = from_substrait_rel(
+                                ctx,
+                                &relations.clone().unwrap_or_default(),
+                                extensions,
+                            )
+                            .await?;
+                            let outer_ref_columns = plan.all_out_ref_exprs();
+                            Ok(Arc::new(Expr::Exists(Exists::new(
+                                Subquery {
+                                    subquery: Arc::new(plan),
+                                    outer_ref_columns,
+                                },
+                                false,
+                            ))))
+                        }
+                        other_type => Err(DataFusionError::Substrait(format!(
+                            "unimplemented type {:?} for set predicate",
+                            other_type
+                        ))),
```

Comment:
would 
```suggestion
                        other_type => substrait_datafusion_err!(
                            "unimplemented type {:?} for set predicate",
                            other_type
                        ),
```
work here as well?